plugins/Live/Controller.php

```diff
@@ -79,6 +79,39 @@ public function getLastVisitsStart()
         $view->idSite = $this->idSite;
         $api = new Request("method=Live.getLastVisitsDetails&idSite={$this->idSite}&filter_limit=10&format=php&serialize=0&disable_generic_filters=1");
         $visitors = $api->process();
+        $visits = array();
```

Comment:
It took me quite a while to review and understand the code. Maybe you can extract this into maybe two methods or so? Something like `getMostUsedUrlsFromVisitors($visitors)` and `assignColorsToUrls($urls)`?

Also the naming confused me a little. Eg `$visits` does not actually contain visits but a count of how often each URL was used. Also we have `$folders` which contain actually URLs? Not sure :)